pkg/crd/gen.go

```diff
@@ -182,6 +189,59 @@ func toTrivialVersions(crd *apiextlegacy.CustomResourceDefinition) {
 	crd.Spec.AdditionalPrinterColumns = canonicalColumns
 }
 
+// removeDefaults removes defaults from apiextensions.k8s.io/v1beta1 CRD definition.
+// To use defaulting, CustomResourceDefinitions must use API version apiextensions.k8s.io/v1
+func removeDefaults(schema *apiextlegacy.JSONSchemaProps) {
```

Comment:
we've got a schema visitor for this purpose, so you shouldn't have to manually write a schema visitor.